src/mono/wasm/debugger/BrowserDebugProxy/DebugStore.cs

```diff
@@ -1183,18 +1183,29 @@ internal SourceFile(AssemblyInfo assembly, int id, DocumentHandle docHandle, Uri
             this.doc = assembly.pdbMetadataReader.GetDocument(docHandle);
             this.docHandle = docHandle;
             this.url = url;
-            this.DebuggerFileName = url.Replace("\\", "/").Replace(":", "");
+
+            this.DebuggerFileName = EscapeAscii(url.Replace("\\", "/"));
             this.BreakableLines = new List<int>();
 
-            this.SourceUri = new Uri((Path.IsPathRooted(url) ? "file://" : "") + url, UriKind.RelativeOrAbsolute);
-            if (SourceUri.IsFile && File.Exists(SourceUri.LocalPath))
-            {
-                this.Url = this.SourceUri.ToString();
-            }
+
+            this.SourceUri = new Uri("file://" + DebuggerFileName, UriKind.RelativeOrAbsolute);
+            if (File.Exists(url))
+                this.Url = "file:///" + DebuggerFileName;
             else
-            {
                 this.Url = DotNetUrl;
+        }
+
+        private static string EscapeAscii(string path)
+        {
+            var builder = new StringBuilder();
+            foreach (var part in Regex.Split(path, @"([:/])"))
```

Comment:
The `:` is needed only for windows. And for !windows, we can do a regular string split on `/`.
If we must use regex here, then maybe move it to a static readonly field.